lib/msf/core/exploit/remote/browser_profile_manager.rb

```diff
@@ -0,0 +1,67 @@
+require 'msgpack'
+
+module Msf
+  module Exploit::Remote::BrowserProfileManager
+
+    # @overload note_type_prefix
+    #  Sets the note type prefix to retrieve or load target information.
+    def note_type_prefix
+      raise NoMethodError, "A mixin that's using BrowserProfileManager should define note_type_prefix"
+    end
+
+
+    # Returns profile information about a specific browser client.
+    #
+    # @param [String] tag A tag that's unique to the browser. Probably generated by Msf::Exploit::Remote::BrowserExploitServer#retrieve_tag.
+    # @return [Hash]
+    def get_profile_info(tag)
+      normalized_tag = "#{note_type_prefix}.#{tag}"
+      framework.db.notes.each do |note|
+        return MessagePack.unpack(note.data) if note.ntype == normalized_tag 
```

Comment:
The call to ``framework.db.notes.each`` is a performance killer on its own. I think in-memory is going to be only feasible option. We need to stop using the database like a scratch pad.